src/models/NetworkAppController.js

```diff
@@ -132,13 +136,16 @@ export default class NetworkAppController {
   }
 
   async uploadContracts() {
-    // TODO: Store the implementation's hash or full source code to avoid unnecessary deployments
+    // TODO: Store the implementation's hash or full source/byte code to avoid unnecessary deployments
     return Promise.all(_.map(this.package.contracts, async (contractName, contractAlias) => {
       const contractClass = ContractsProvider.getFromArtifacts(contractName);
       log.info(`Uploading ${contractName} implementation for ${contractAlias}`);
       const contractInstance = await this.appManagerWrapper.setImplementation(contractClass, contractAlias);
       log.info(`Uploaded ${contractName} at ${contractInstance.address}`);
-      this.networkPackage.contracts[contractAlias] = contractInstance.address;
+      this.networkPackage.contracts[contractAlias] = {
+        address: contractInstance.address,
+        bytecode: contractClass.bytecode
```

Comment:
what about storing a hash of the bytecode here instead of storing all of it? it will improve the readability of the json file

Reply:
I was in doubt about that. On one hand it does improve readability, on the other it allows you to reconstruct a deployment from scratch _without_ needing to have a separate json file with the bytecode. However, we are not using it for anything for now.